gulpfile.js

```diff
@@ -124,28 +172,197 @@ gulp.task("default", function (callback) {
     runSequence("clean", "compile", "tslint", callback);
 });
 
-gulp.task("test-ios", function (callback) {
-    runSequence("test-ios-uiwebview", "test-ios-wkwebview", callback);
-});
+function startEmulators(callback, restartIfRunning, android, ios) {
+    // called when an emulator is initialized successfully
+    var emulatorsInit = 0;
+    function onEmulatorInit(emulator) {
+        ++emulatorsInit;
+        console.log(emulator + " emulator is ready!");
+        if (emulatorsInit === ((android ? 1 : 0) + (ios ? 1 : 0))) {
+            console.log("All emulators are ready!");
+            callback(undefined);
+        }
+    }
+    
+    // called to check if an Android emulator is initialized
+    function androidEmulatorReady(onFailure) {
+        console.log("Checking if Android emulator is ready yet...");
+        // dummy command that succeeds if emulator is ready and fails otherwise
+        spawnCommand("adb", ["shell", "pm", "list", "packages"], (code) => {
```

Comment:
Lets stick to ES5 syntax (no arrow functions) for this gulpfile since 1. It's the convention throughout this file and 2. It won't work on older versions of node.